sparse/unit_test/Test_Sparse_removeCrsMatrixZeros.hpp

```diff
@@ -0,0 +1,193 @@
+//@HEADER
+// ************************************************************************
+//
+//                        Kokkos v. 4.0
+//       Copyright (2022) National Technology & Engineering
+//               Solutions of Sandia, LLC (NTESS).
+//
+// Under the terms of Contract DE-NA0003525 with NTESS,
+// the U.S. Government retains certain rights in this software.
+//
+// Part of Kokkos, under the Apache License v2.0 with LLVM Exceptions.
+// See https://kokkos.org/LICENSE for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//@HEADER
+
+/// \file Test_Sparse_SortCrs.hpp
+/// \brief Tests for sort_crs_matrix and sort_crs_graph in
+/// KokkosSparse_SortCrs.hpp
+
+#ifndef KOKKOSSPARSE_REMOVECRSZEROS_HPP
+#define KOKKOSSPARSE_REMOVECRSZEROS_HPP
+
+#include <Kokkos_Core.hpp>
+#include <KokkosSparse_CrsMatrix.hpp>
+#include <KokkosSparse_Utils.hpp>
+#include <Kokkos_ArithTraits.hpp>
+
+namespace TestRemoveCrsMatrixZeros {
+
+// Simple, sequential implementation of zero-removal to compare against
+template <typename Matrix>
+Matrix removeMatrixZerosReference(const Matrix& A) {
+  using Offset  = typename Matrix::non_const_size_type;
+  using Ordinal = typename Matrix::ordinal_type;
+  using Scalar  = typename Matrix::value_type;
+  using KAT     = Kokkos::ArithTraits<Scalar>;
+  auto rowmapHost =
+      Kokkos::create_mirror_view_and_copy(Kokkos::HostSpace(), A.graph.row_map);
+  auto entriesHost =
+      Kokkos::create_mirror_view_and_copy(Kokkos::HostSpace(), A.graph.entries);
+  auto valuesHost =
+      Kokkos::create_mirror_view_and_copy(Kokkos::HostSpace(), A.values);
+  // First, create the filtered rowmap (the CrsMatrix constructor taking host
+  // pointers does expect rowmap to be in Ordinal)
+  std::vector<Ordinal> filteredRowmap;
+  Offset filteredNNZ = 0;
+  for (Ordinal i = 0; i <= A.numRows(); i++) {
```

Comment:
Nothing wrong here but it seems more straightforward as follows since it avoids the `<=` and `break` checks
```
Offset filteredNNZ = 0;
filteredRowmap.push_back(filteredNNZ);
for(Ordinal i = 0; i < A.numRows(); i++) {
  for(Offset j = rowmapHost(i); j < rowmapHost(i+1); ++j) {
    if(valuesHost(j) != KAT::zero()) {
      ++filteredNNZ;
    }
  }
  filteredRowmap.pushback(filteredNNZ);
}
```